src/lib/InstantSearch.js

```diff
@@ -90,6 +90,31 @@ Usage: instantsearch({
     this.widgets.push(widget);
   }
 
+  removeWidget(widget) {
+    if (
+      !this.widgets.includes(widget) ||
+      typeof widget.dispose !== 'function'
+    ) {
+      throw new Error('Widget definition is missing dispose method');
```

Comment:
I would go for something like "The widget you tried to remove does not implement the `dispose` method, therefore it is not possible to remove this widget" in order to provide more information and make it less abstract.